File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/crc32.c
local void braid(z_crc_t ltl[][256], z_word_t big[][256], int n, int w) {
    int k;
    z_crc_t i, p, q;
    for (k = 0; k < w; k++) {
        p = x2nmodp((n * w + 3 - k) << 3, 0);
        ltl[k][0] = 0;
        big[w - 1 - k][0] = 0;
        for (i = 1; i < 256; i++) {
            ltl[k][i] = q = multmodp(i << 24, p);
            big[w - 1 - k][i] = byte_swap(q);
        }
    }
}

File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/crc32.rs
rust
fn braid<const N: usize, const W: usize>(ltl: [[z_crc_t; 256]; W], big: [[z_word_t; 256]; W]) {
    for k in 0..W {
        let p = x2nmodp((N * W + 3 - k) << 3, 0);
        ltl[k][0] = 0;
        big[W - 1 - k][0] = 0;
        for i in 1..=256 {
            ltl[k][i] = multmodp(i << 24, p);
            big[W - 1 - k][i] = byte_swap(ltl[k][i]);
        }
    }
}